playbooks/init/evaluate_groups.yml

```diff
@@ -146,7 +146,11 @@
       groups: oo_nodes_to_bootstrap
       ansible_ssh_user: "{{ g_ssh_user | default(omit) }}"
       ansible_become: "{{ g_sudo | default(omit) }}"
-    with_items: "{{ groups.oo_masters | default([]) }}"
+    with_items: "{{ g_new_master_hosts | default(g_master_hosts | default([], true), true) }}"
+    when:
+    # Add masters to oo_nodes_to_bootstrap when we are not scaling up.
+    # Add new masters to oo_nodes_to_bootstrap when we are scaling up masters.
+    - g_new_node_hosts | length == 0 or (g_new_node_hosts | length > 0 and g_new_master_hosts | length > 0)
```

Comment:
g_new_node_hosts | length > 0 is redundant to the first condition.  It's never going to be negative, so it can only be zero or greater than zero.  g_new_master_hosts implies new_nodes as well.

Other than that, this looks good to me.